subsys/random/rand32_entropy_device.c

```diff
@@ -86,14 +86,9 @@ static int rand_get(uint8_t *dst, size_t outlen, bool csrand)
 
 		while (len < outlen) {
 			random_num = k_cycle_get_32();
-			if ((outlen-len) < sizeof(random_num)) {
-				blocksize = len;
-				(void)memcpy(&(dst[random_num]),
-						&random_num, blocksize);
-			} else {
-				*((uint32_t *)&dst[len]) = random_num;
-			}
-
+			(void)memcpy(&(dst[len]), &random_num,
+				(outlen - len > blocksize) ? blocksize :
+				outlen - len);
 			len += blocksize;
```

Comment:
While the above works IMO it would be more clear to do something like:

    size_t copylen = outlen - len;

    if (copylen > blocksize) {
        copylen = blocksize;
    }
    (void)memcpy(&dst[len], &random_num, copylen);
    len += copylen;

because adding `blocksize` when only `outlen - len` was copied is confusing.

Reply:
I don't mind change, but both ways would just finish the loop that is why I didn't bother changing it

Reply:
just done :)

Reply:
Thanks.

As I said, it works, it just means `len` is not the number of bytes copied into `dst` as it is throughout the loop up to the last point.  So somebody who needed to modify this to use `len` after the loop could waste a lot of time figuring out it was wrong.

I don't think it's good to make it harder for people to maintain and evolve code.  That's not a sign of a quality software system.